python/ml4ir/base/model/optimizer.py

```diff
@@ -37,38 +24,83 @@ def get_optimizer(
     References:
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
+        https://arxiv.org/pdf/1506.01186.pdf
 
     FIXME:
         Define all arguments overriding tensorflow defaults in a separate file
         for visibility with ml4ir users
     """
-    # Define an exponential learning rate decay schedule
-    learning_rate_schedule = ExponentialDecay(
-        learning_rate,
-        decay_steps=learning_rate_decay_steps,
-        decay_rate=learning_rate_decay,
-        staircase=True,
-    )
+    model_config = file_io.read_yaml(model_config_file)
```

Comment:
Do we need to re-read the config file? We could save passing the `file_io`. I am thinking in a notebook for example, to do this, you may want to pass a dict but passing a file_io and a file_pointer does not work well.

Reply:
That's a good point. I think the way thing are currently defined, you need to read it twice because the Scorer uses the model config to create the hidden neural network architecture. The RelevanceModel uses the model_config to configure the optimizer. I think the best way forward is to read it in the pipeline, but still have functionality to read from file in the individual methods if needed. Check this -> https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/model/scoring/scoring_model.py#L64